java-client-model/src/main/java/com/cumulocity/model/idtype/GId.java

```diff
@@ -57,15 +58,21 @@ public String toJSON() {
         return JSON.defaultJSON().quote(super.getValue());
     }
 
+    /**
+     * GId is not always a number anymore
+     * The method will be removed with a next release
+     */
+    @Deprecated
     public Long getLong() {
+        if (getValue() == null) {
+            return null;
+        }
         try {
-            if (getValue() == null) {
-                return null;
-            }
             return Long.parseLong(getValue());
         } catch (Exception e) {
-            return null;
+            Throwables.propagateIfPossible(e, NumberFormatException.class);
```

Comment:
Do we need this exception? This is potential breaking change, if we are deprecating whole method I would avoid changing it behavior if possible.

Reply:
But we know that id is no longer a number, so IMO it should fail fast, cause it will cause hard to find issues